subsys/sd/sd_ops.c

```diff
@@ -63,21 +63,20 @@ int sdmmc_read_status(struct sd_card *card)
 int sdmmc_wait_ready(struct sd_card *card)
 {
 	int ret, timeout = CONFIG_SD_DATA_TIMEOUT * 1000;
-	bool busy = true;
 
 	do {
-		busy = sdhc_card_busy(card->sdhc);
-		if (!busy) {
+		if (!sdhc_card_busy(card->sdhc)) {
 			/* Check card status */
 			ret = sd_retry(sdmmc_read_status, card, CONFIG_SD_RETRY_COUNT);
-			busy = (ret != 0);
-		} else {
-			/* Delay 125us before polling again */
-			k_busy_wait(125);
-			timeout -= 125;
+			if (ret == 0) {
+				return 0;
+			}
 		}
-	} while (busy && (timeout > 0));
-	return busy;
+		/* Delay 125us before polling again */
+		k_busy_wait(125);
+		timeout -= 125;
+	} while (timeout > 0);
+	return 1;
```

Comment:
What about returning `-EBUSY`? The comment for `sdmmc_wait_ready` indicates it should return 0 if the card is ready (to indicate the wait succeeded).

Reply:
Change applied. I've also updated the timeout logic, so that the timeout duration is a bit closer to `CONFIG_SD_DATA_TIMEOUT` in the case where the card is detached.